src/h3lib/lib/h3Index.c

```diff
@@ -471,21 +471,21 @@ int H3_EXPORT(compactCells)(const H3Index* h3Set, H3Index* compactedSet,
  * @return              An error code if output array is too small or any cell
  *                      is smaller than the output resolution.
  */
-int H3_EXPORT(uncompactCells)(const H3Index* compactedSet,
-                              const int64_t numCompacted, H3Index* outSet,
-                              const int64_t numOut, const int res) {
+H3Error H3_EXPORT(uncompactCells)(const H3Index* compactedSet,
+                                  const int64_t numCompacted, H3Index* outSet,
+                                  const int64_t numOut, const int res) {
     int64_t i = 0;
 
     for (int64_t j = 0; j < numCompacted; j++) {
-        if (!_hasChildAtRes(compactedSet[j], res)) return -2;
+        if (!_hasChildAtRes(compactedSet[j], res)) return E_FAILED;
```

Comment:
I used the E_RES_MISMATCH for these cases, since it covers the H3Index res being invalid. There is a bit of overlap between E_RES_MISMATCH (the index resolution is out of range) and E_RES_DOMAIN (the resolution argument itself is out of range), which is tricky.